File: DSA/Mayank_590018496/Day_1_Question_2.c
#include <stdio.h>

void find_missing_and_repeated(int nums[], int n, int *missing, int *repeated) {
    int freq[n + 1];
    
    for (int i = 1; i <= n; i++) {
        freq[i] = 0;
    }

    for (int i = 0; i < n; i++) {
        freq[nums[i]]++;
    }

    for (int i = 1; i <= n; i++) {
        if (freq[i] == 0) {
            *missing = i;
        } else if (freq[i] > 1) {
            *repeated = i;
        }
    }
}

int main() {
    int n;
    
    printf("Enter the number of elements: ");
    scanf("%d", &n);

    int nums[n];
    
    printf("Enter %d elements: ", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &nums[i]);
    }

    int missing, repeated;
    find_missing_and_repeated(nums, n, &missing, &repeated);

    printf("Repeated number: %d\n", repeated);
    printf("Missing number: %d\n", missing);

    return 0;
}

File: DSA/ARARTIK_TYAGI_590011633/DAY_1_QUESTION_2.py
def find_missing_and_repeated(nums):
    n = len(nums)
    expected_sum = n * (n + 1)  // 2
    actual_sum = sum(nums)
    seen = set()
    repeated = -1

    for num in nums:
        if num in seen:
            repeated = num
        else:
            seen.add(num)

    missing = expected_sum - (actual_sum - repeated)
    return (missing, repeated)

#Input user 
nums = list(map(int, input("Enter the array elements (1 to n, one repeated, one missing): ").split()))
missing, repeated = find_missing_and_repeated(nums)
print("Missing: ", missing)
print("Repeated:", repeated)